### DISCUSSION TITLE
Make categories for adding random meals configurable

### DISCUSSION THREAD

**Discussion by Belgadon** [mealie-recipes/mealie#2304] [Feature Request]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [X] Reviewed the documentation and verified that my feature request isn't already a feature.
- [X] Reviewed the existing feature requests and my feature has not been requested.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
Adding random recipes to the meal plan could be imporoves in two ways
a) Breakfast and lunch are behind an extra icon/extra click, dinner and sides are not.
b) All buttons/icons add a random recipe.

**(Optional): Suggest A Solution**
a) The buttons/icons should be consistent. Make all options clickable with one click only.
b) Make it configurable what categories should be use when clicking the icons/buttons. e.g. icon "Random Lunch" adds recipes with the category "Main Dish"

**Additional Information**
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by michael-genson** [comment]
> Make it configurable what categories should be use when clicking the icons/buttons. e.g. icon "Random Lunch" adds recipes with the category "Main Dish"

I believe you're describing meal plan rules, which does exactly this:

![image](https://github.com/user-attachments/assets/8f178efe-f8e8-4a19-b8ef-90cbd3e560bc)

If this is incorrect, feel free to re-open with additional context
